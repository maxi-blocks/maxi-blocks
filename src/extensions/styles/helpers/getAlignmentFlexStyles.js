/**
 * General
 */
const breakpoints = ['general', 'xxl', 'xl', 'l', 'm', 's', 'xs'];

/**
 * External dependencies
 */
import { isEmpty } from 'lodash';

const getAlignmentFlexStyles = obj => {
	const response = {};

	breakpoints.forEach(breakpoint => {
		if (!isEmpty(obj[`alignment-${breakpoint}`]))
			switch (obj[`alignment-${breakpoint}`]) {
				case 'left':
					response[breakpoint] = {
						'justify-content': 'flex-start',
					};
					break;
				case 'center':
				case 'justify':
					response[breakpoint] = {
						'justify-content': 'center',
					};
					break;
				case 'right':
					response[breakpoint] = {
						'justify-content': 'flex-end',
					};
					break;
				default:
					break;
			}
	});

	return response;
};

export default getAlignmentFlexStyles;
