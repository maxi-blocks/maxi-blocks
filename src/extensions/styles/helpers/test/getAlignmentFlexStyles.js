import getAlignmentFlexStyles from '../getAlignmentFlexStyles';

describe('getAlignmentFlexStyles', () => {
	it('Get a correct alignment flex styles', () => {
		const object = {
			'alignment-general': 'right',
			'alignment-xxl': 'left',
			'alignment-xl': 'right',
			'alignment-l': 'left',
			'alignment-m': 'right',
			'alignment-s': 'left',
			'alignment-xs': 'right',
		};

		const result = getAlignmentFlexStyles(object);
		expect(result).toMatchSnapshot();
	});
});
