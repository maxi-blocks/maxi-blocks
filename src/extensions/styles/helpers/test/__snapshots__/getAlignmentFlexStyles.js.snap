// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAlignmentFlexStyles Get a correct alignment flex styles 1`] = `
{
  "general": {
    "justify-content": "flex-end",
  },
  "l": {
    "justify-content": "flex-start",
  },
  "m": {
    "justify-content": "flex-end",
  },
  "s": {
    "justify-content": "flex-start",
  },
  "xl": {
    "justify-content": "flex-end",
  },
  "xs": {
    "justify-content": "flex-end",
  },
  "xxl": {
    "justify-content": "flex-start",
  },
}
`;
